test/e2e/e2eutil/wait_util.go

```diff
@@ -0,0 +1,144 @@
+package e2eutil
```

Comment:
We do have a another wait related method `waitUntilSizeReached` that will be refactored here soon. I just wanted to logically group all the wait and retry type methods together.
Or we can rename it if you think that's better.